lib/redis_client/cluster.rb

```diff
@@ -16,42 +16,47 @@ class Cluster
     def initialize(config, pool: nil, concurrency: nil, **kwargs)
       @config = config
       @concurrent_worker = ::RedisClient::Cluster::ConcurrentWorker.create(**(concurrency || {}))
-      @router = ::RedisClient::Cluster::Router.new(config, @concurrent_worker, pool: pool, **kwargs)
       @command_builder = config.command_builder
+
+      @pool = pool
+      @kwargs = kwargs
+      @router = nil
+      @mutex = Mutex.new
     end
 
     def inspect
-      "#<#{self.class.name} #{@router.node_keys.join(', ')}>"
+      node_keys = @router.nil? ? @config.startup_nodes.keys : router.node_keys
```

Comment:
Hi! 👋

I'm investigating the behavior of the redis-clustering gem and came across this PR.
I have a question about this specific method:

Why does this method directly access the `@router` variable instead of using the `router` method like other methods? 
It seems that calling the `router` method would initialize `@router`.

Is there a specific reason for this approach? Any insights would be greatly appreciated!

Thanks in advance for your help!

Reply:
Since the inspect method doesn't call a command, I thought it shouldn't be done the initialization.